src/vmm/src/builder.rs

```diff
@@ -634,26 +628,28 @@ fn attach_net_devices(
 
         let tap = cfg.open_tap().map_err(|_| NetDeviceNotConfigured)?;
 
-        let net_box = Arc::new(Mutex::new(
-            devices::virtio::Net::new_with_tap(
-                tap,
-                cfg.guest_mac(),
-                epoll_config,
-                rx_rate_limiter,
-                tx_rate_limiter,
-                allow_mmds_requests,
+        let net_device = event_manager
+            .register(
+                devices::virtio::Net::new_with_tap(
+                    tap,
+                    cfg.guest_mac(),
+                    vmm.guest_memory().clone(),
+                    rx_rate_limiter.unwrap_or_default(),
+                    tx_rate_limiter.unwrap_or_default(),
+                    allow_mmds_requests,
+                )
+                .map_err(CreateNetDevice)?,
             )
-            .map_err(CreateNetDevice)?,
-        ));
+            .map_err(StartMicrovmError::RegisterEvent)?;
 
         attach_mmio_device(
             vmm,
             cfg.iface_id.clone(),
-            MmioTransport::new(vmm.guest_memory().clone(), net_box)
-                .map_err(device_manager::mmio::Error::CreateMmioDevice)
-                .map_err(RegisterNetDevice)?,
+            MmioTransport::new(vmm.guest_memory().clone(), net_device.clone()).map_err(|e| {
```

Comment:
No need to `clone` the `net_device` here.

Reply:
nit: remove this `.clone()`

Reply:
Done. Please take a look at the last force push.